packages/web/src/components/upload/UploadChip.tsx

```diff
@@ -30,58 +51,112 @@ type UploadChipProps = {
   /**
    * Is this upload the user's first of this type
    * */
+  source: 'nav' | 'profile' | CreatePlaylistSource
   isFirst?: boolean
-  onClick: () => void
-  isArtist?: boolean
 }
 
 const UploadChip = ({
   type = 'track',
   variant = 'tile',
-  isArtist = false,
   isFirst = false,
-  onClick
+  source
 }: UploadChipProps) => {
+  const { isEnabled: isEditAlbumsEnabled } = useFlag(FeatureFlags.EDIT_ALBUMS)
+  const messages = getMessages(type)
   const icon =
-    type === 'track' || type === 'album' ? (
+    type === 'track' ? (
       <IconUpload className={styles.iconUpload} />
     ) : (
-      <IconMultiselectAdd className={styles.iconPlus} />
+      <IconPlus className={styles.iconPlus} color='subdued' />
     )
 
-  let text
+  let text: string
   switch (type) {
     case 'track':
       text = variant === 'nav' ? messages.track : messages.aTrack
       break
     case 'album':
-      text = isFirst ? messages.firstAlbum : messages.album
-      break
     case 'playlist':
-      if (isArtist) {
-        text = isFirst ? messages.firstArtistPlaylist : messages.artistPlaylist
-      } else {
-        text = isFirst ? messages.firstPlaylist : messages.playlist
-      }
+      text = messages.createNewCollection(isFirst)
       break
     default:
       break
   }
 
-  return (
+  const dispatch = useDispatch()
+
+  const handleClickUpload = useCallback(() => {
+    dispatch(pushRoute(UPLOAD_PAGE))
+    track(
+      make({
+        eventName: Name.TRACK_UPLOAD_OPEN,
+        source: source as 'nav' | 'profile'
+      })
+    )
+  }, [dispatch, source])
+
+  const handleCreateCollection = useCallback(() => {
+    dispatch(
+      (type === 'album' ? createAlbum : createPlaylist)(
+        { playlist_name: `New ${type}` },
+        source as CreatePlaylistSource,
+        undefined,
+        'route'
+      )
+    )
+  }, [dispatch, source, type])
+
+  const items: PopupMenuItem[] = useMemo(
+    () =>
+      type === 'track'
+        ? []
+        : [
+            {
+              text: messages.uploadCollection,
+              icon: <IconCloudUpload />,
+              onClick: handleClickUpload
+            },
+            {
+              text: messages.createCollection,
+              icon: <IconPlaylists />,
+              onClick: handleCreateCollection
```

Comment:
ah seems like it would maybe be hard to make these links. something to shoot for with harmony popup/menu